### PR TITLE
Fix test reports, output variables, misc fixes

### PR DESCRIPTION
**Pull request by vistaarjuneja**
i) Added test reports support
ii) Renamed everywhere to harness-docker-runner instead of "lite-engine"
iii) fix output variables

### CODE REVIEW

**Review by happyhebaby — APPROVED** [review]

File: ti/callgraph/upload.go [hunk 1]
```diff
@@ -26,11 +26,11 @@ const (
 )
 
 // Upload method uploads the callgraph.
-func Upload(ctx context.Context, stepID string, timeMs int64, out io.Writer) error {
+func Upload(ctx context.Context, stepID string, timeMs int64, out io.Writer, ticlient client.Client) error {
 	log := logrus.New()
 	log.Out = out
 
-	// TODO: Remove cyclic dependency
+	// TODO: Pass in the config here and use that, right now it will be empty
```
Comment (happyhebaby) [inline ti/callgraph/upload.go:33]: What blocks to include ti config now? I thought we could retrieve the info from the saved mapping info?

File: pipeline/runtime/runtest.go [hunk 2]
```diff
@@ -45,11 +46,12 @@ func executeRunTestStep(ctx context.Context, engine *engine.Engine, r *api.Start
 	log.Out = out
 
 	exited, err := engine.Run(ctx, step, out)
-	if rerr := report.ParseAndUploadTests(ctx, r.TestReport, r.WorkingDir, step.Name, log); rerr != nil {
+	// TODO: Pass in the TI client here as well
```
Comment (happyhebaby) [inline pipeline/runtime/runtest.go:49]: This comment should be removed?

File: ti/callgraph/upload.go (hunk 1 above)
Comment (vistaarjuneja) [reply ti/callgraph/upload.go:33]: We are only passing in the client at the moment, we need to do the same for the TI Config (only used for RunTests step)

File: pipeline/runtime/runtest.go (hunk 2 above)
Comment (vistaarjuneja) [reply pipeline/runtime/runtest.go:49]: yeah, will remove

**Pull request merged**